File: ClientProject/Assets/.lpproj/Apis/Unity/CS/GameobjectUtils.lua
---@class CS.GameobjectUtils : CS.System.Object
CS.GameobjectUtils = {}

---@return CS.GameobjectUtils
function CS.GameobjectUtils()
end

---@param go : CS.UnityEngine.GameObject
---@param x : CS.System.Single
---@param y : CS.System.Single
---@param z : CS.System.Single
function CS.GameobjectUtils.SetWorldPos(go, x, y, z)
end

---@param go : CS.UnityEngine.GameObject
---@return CS.UnityEngine.Vector3
function CS.GameobjectUtils.GetWorldPos(go)
end

---@param go : CS.UnityEngine.GameObject
---@param x : CS.System.Single
---@param y : CS.System.Single
---@param z : CS.System.Single
function CS.GameobjectUtils.SetLocalPos(go, x, y, z)
end

---@param go : CS.UnityEngine.GameObject
---@return CS.UnityEngine.Vector3
function CS.GameobjectUtils.GetLocalPos(go)
end

---@param go : CS.UnityEngine.GameObject
---@param x : CS.System.Single
---@param y : CS.System.Single
---@param z : CS.System.Single
function CS.GameobjectUtils.SetLocalScale(go, x, y, z)
end

---@param go : CS.UnityEngine.GameObject
---@param parent : CS.UnityEngine.Transform
function CS.GameobjectUtils.SetParent(go, parent)
end

---@param go : CS.UnityEngine.GameObject
---@param name : CS.System.String
---@return CS.UnityEngine.Transform
function CS.GameobjectUtils.FindChild(go, name)
end

---@param go : CS.UnityEngine.GameObject
---@param index : CS.System.Int32
---@return CS.UnityEngine.Transform
function CS.GameobjectUtils.GetChild(go, index)
end

---@param go : CS.UnityEngine.GameObject
---@return CS.System.Int32
function CS.GameobjectUtils.GetChildCount(go)
end

File: ClientProject/Assets/CSharpScript/Utils/GameobjectUtils.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class GameobjectUtils
{

	public static void SetWorldPos(GameObject go, float x, float y, float z)
    {
        if (go != null)
        {
            go.transform.position = new Vector3(x, y, z);
        }
    }

    public static Vector3 GetWorldPos(GameObject go)
    {
        if (go != null)
        {
            return go.transform.position;
        }
        return Vector3.zero;
    }

    public static void SetLocalPos(GameObject go, float x, float y, float z)
    {
        if (go != null)
        {
            go.transform.localPosition = new Vector3(x, y, z);
        }
    }

    public static Vector3 GetLocalPos(GameObject go)
    {
        if (go != null)
        {
            return go.transform.localPosition;
        }
        return Vector3.zero;
    }


    public static void SetLocalScale(GameObject go, float x, float y, float z)
    {
        if (go != null)
        {
            go.transform.localScale = new Vector3(x, y, z);
        }
    }

    public static void SetParent(GameObject go, Transform parent)
    {
        if (go != null)
        {
            go.transform.parent = parent;
        }
    }

    public static Transform FindChild(GameObject go, string name)
    {
        if (go != null)
        {
            return go.transform.Find(name);
        }
        return null;
    }

    public static Transform GetChild(GameObject go, int index)
    {
        if (go != null)
        {
            return go.transform.GetChild(index);
        }
        return null;
    }

    public static int GetChildCount(GameObject go)
    {
        if (go != null)
        {
            return go.transform.childCount;
        }
        return 0;
    } 
}
